### PR TITLE
Function advancement data structures

### LINKED ISSUE #15: Create advancement, function data structures

### PR DESCRIPTION
**Pull request by JustinHuPrime**
Added global store object and data structures representing runtime values of functions, advancements, triggers, item specs, etc.

Closes #15

### CODE REVIEW

File: src/codeGenerator/store.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+export class InventoryChanged extends Trigger {
+  item: ItemSpec;
+
+  constructor(item: ItemSpec) {
+    super();
+    this.item = item;
+  }
+}
+
+export class Raw extends Trigger {
```
Comment (jlu5) [inline src/codeGenerator/store.ts:76]:
```suggestion
export class RawTrigger extends Trigger {
```

Maybe this is a more specific name?

**Review by jlu5 — APPROVED** [review]

File: src/codeGenerator/store.ts (hunk 1 above)
Comment (JustinHuPrime) [reply src/codeGenerator/store.ts:76]: No, on the grounds that every other trigger doesn't mention its parent class

File: src/codeGenerator/store.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+export class TagMatcher extends ItemSpec {
+  tag: string;
+
+  constructor(tag: string) {
+    super();
+    this.tag = tag;
+  }
+}
```
Comment (curtiskoo) [inline src/codeGenerator/store.ts:103]: is `TagMatcher` compatible with ConsumeItem trigger? i.e. can we specify a tag in place of an item id for the consume_item trigger?

Comment (JustinHuPrime) [reply src/codeGenerator/store.ts:103]: Yes, definitely.

Comment (curtiskoo) [reply src/codeGenerator/store.ts:103]: interesting, have not tried that before, thanks

File: src/codeGenerator/store.ts (hunk 1 above)
Comment (jlu5) [reply src/codeGenerator/store.ts:76]: I would rather change them all to include Trigger tbh - class names should be clear and self contained, I'm convinced a module name of "Store" alone makes it clear that all of these are triggers.

**Review by curtiskoo — APPROVED** [review]
lgtm

**Pull request merged**